### ISSUE TITLE
Ableton Live Can't Load Driver

### ISSUE THREAD

**Issue by lagauche** [eiz/SynchronousAudioRouter#31]
When selecting the Synchronous driver in Ableton Win64 it says:

The device only has input channels. Live only supports audio devices that have at least one stereo output.

Then the driver fails to load.

Any ideas?

**Comment by MSM792** [comment]
I had a similar issue with Studio One. I think this was due to the previous configuration being incompatible with Studio One (probably because there haven't been any outputs configured). Anyway, Studio One would not load the SynchronousAudioRouter driver and therefore wouldn't allow me to open the control panel. My solutions to this was to delete the folder %AppData%\SynchronousAudioRouter. After that the driver loaded and I could start over with the configuration.

**Comment by dreamcat4** [comment]
Suggested solution did not work for me. Same error msg repeat even after deleting the AppData subfolder. Didn't close ableton live 9. Didnt restart win10.

https://i.imgur.com/2FSkwKy.png

**Comment by paulheu** [comment]
Install Reaper to verify SAR is installed and working correctly

**Comment by dreamcat4** [comment]
@paulheu yes reaper did allow me to select SAR in its options page. And access the SAR control panel from there. I created 2 output channels.

However I had a couple of other issues over there initially trying to use reaper. Firstly it didnt list them in my output list. The main options page of reaper only permitted me to select my 2 'SDIF digital autio out' channels for output first-last. Wheras I wanted to select 2 other analogue output channels. All the system analog channels were missing... ok well maybe i just dont understand how to use the software yet. Then shortly after going back to the main window... well reaper crashed pretty soon after that.

So for those other reason(s)... I was then just struggling with reaper instead. Installing jack didn't get me anywhere because [unlike reaper] it seemed like nothing from SAR appeared in jack. (But I totally dont know how to use jack, and would have 0 success understanding what to do with it anyway, its so hard / tough to use by comparison / has its own problems).

**Comment by dreamcat4** [comment]
OK reaper is now launched (as admin). And now I try creating a SAR input device too. Which appears in reaper output channel section. This is progress! Unfortunately only the windows 1 'selected default playback' device can be selected in reaper device output at any one time. Afaikt this is an issue of windows audio systems. But I would like to find a way to work around it.

For example if my back surround channels are piped into the SAR input, maybe that can be linked using jackd.exe to a different [real] other windows playback device. That is not the currently selected 'default playback device'.

Or maybe I can simultaneously run 2 different DAW software. With 'daw A' outputting to 'real audio playback device A'. And 'daw B' outputting to 'real audio playback device B'. Or this is what 2 instances of jackd.exe can do with the right flags.

Also new issue now with win10. Is that whenever I l*try* aunch reaper as 'admin'... well the explorer.exe crashes every other (alternate time). And reaper doesn't get started. That is some other bug. So to launch reaper once requires me to launch it twice. And get past that initial crashing.

OK so now the SAR created inputs and outputs are both showing up in the Reaper main options page. And can be included in the available inputs-outputs channels device ranges. However when I click 'OK' to save options.... reaper just crashes again anyway. Therefore reaper its still really unstable for me here on windows 10. When I click 'debug', the 'reaper' crash seemed to occured in a windows audio dll (api call) being used by reaper:

https://i.imgur.com/JSyJivR.png

I guess then (for those other problems encountered elsewhere)... its still really nice alternative to have SAR working for Ableton too. It is useful 2nd program to see / compare and identify if a bug or problem is either in the DAW, or is in SAR, or in windows 10 audio drivers stack.

Hence this ticket.

**Comment by dreamcat4** [comment]
Ah okay!

Then going to reaper is a work-around. We just added some virtual devices from launching SAR config in reaper, then exit reaper. Or reaper just crashes anyhow (whatever, it doesnt matter). Now I can launch ableton, and it no longer complains! The error message is gone, because SAR has got some ins/outs now!

This is great. Another thing though: the daw is spinning wheel now. For example after saving the SAR config. Or selecting 'input channels' and 'output channels'. A toast notification comes up for 'ASIO v2'. Saying the audio mode or whatever. And then its spinning wheel in Ableton. What is better: it isnt full out crashing (seem Ableton is too good for that, and has some way to recover itself). But its definately hanging for several seconds at these times. Still buggy / problems occur, which is not the fault of the DAW it seems. So win10 build 15063 and SAR + ASIO4ALLv2 together is not playing so well....

Maybe I should see if there is an update for ASIO4ALLv2... nope its the newest version (23 May 2017: Version 2.14).

**Comment by jontzi** [comment]
@dreamcat4 could you send your Ableton compatible default.json config file from %appdata%\SynchronousAudioRouter folder? i think issue with Ableton is that the default config is not compatible with Ableton and you need to use other DAW to config it first as Reaper seems to ignore lack of audio channels configured by default. :)

**Comment by jontzi** [comment]
@dreamcat4 nevermind, got it working by doing setup via Reaper first :)

here's working config for ableton (insert in %APPDATA%\SynchronousAudioRouter\default.json file):

`
{
  "applications": [
    {
      "description": "Ableton",
      "path": "C:\\ProgramData\\Ableton\\Live 9 Suite\\Program\\Ableton Live 9 Suite.exe"
    }
  ],
  "driverClsid": "{429A7462-F176-411F-8DA9-AE6C0A258098}",
  "enableApplicationRouting": true,
  "endpoints": [
    {
      "channelCount": 2,
      "description": "",
      "id": "ep_1",
      "type": "playback"
    },
    {
      "channelCount": 2,
      "description": "",
      "id": "ep_2",
      "type": "recording"
    }
  ]
}
`

**Comment by r0zar** [comment]
FIX FOR ABLETON USERS:

1. Close Ableton
2. Rename your %APPDATA%\SynchronousAudioRouter\default.json to defaultx.json
3. Start Ableton and load the driver again.

basically it's get into a bad state and needs to be reset. cheers!
